### PR TITLE
PRO-1727 Social Logins Onboarding

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1727/pillar-wallet-social-logins-onboarding

## Description

- Implemented 6 social sign-in using web3auth.

## Motivation and Context

-

## How Has This Been Tested?

- Tested on Simulator 14 PRO (Testnet and Mainnet)
- Tested on Android device (Testnet and Mainnet)

## Screenshots (if appropriate):
https://github.com/pillarwallet/pillarwallet/assets/82652040/39360bcf-8dcc-42d0-ba05-8b98e3db93e9

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by Yash-Chaubal — APPROVED** [review]
LGTM! Just a couple of questions, but should be good to merge afterwards

File: src/configs/buildConfig.js [hunk 1]
```diff
@@ -37,6 +37,10 @@ export const buildEnvironment = {
   KOCHAVA_IOS_ID: '_kochava_ios_id_',
   KOCHAVA_ANDROID_ID: '_kochava_android_id_',
   WALLETCONNECT_PROJECT_ID: '_wallet_connect_id_',
+  // eslint-disable-next-line i18next/no-literal-string
+  WEB3_AUTH_CLIENT_ID: '_web3_auth_client_id_',
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/configs/buildConfig.js:41]: Screenshot 2023-08-18 at 2 38 38 PM

File: ios/Podfile.lock [hunk 2]
```diff
@@ -941,8 +949,10 @@ SPEC CHECKSUMS:
...
   toolbar-android: 2a73856e98b750d7e71ce4644d3f41cc98211719
+  Web3Auth: 83062b2ccc9b5882f44c02fa3464283f833c169f
+  web3auth-react-native-sdk: f3bf00cde3cc93e30895a03499dbcb26739307a2
   Yoga: 4bd86afe9883422a7c4028c00e34790f560923d6
 
-PODFILE CHECKSUM: 9c66e94900b997268a39f6560cad896ff3791b27
+PODFILE CHECKSUM: 83a49c75f95ef8ab38dce778342c3ba6af89ba27
 
-COCOAPODS: 1.12.1
+COCOAPODS: 1.11.3
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply ios/Podfile.lock]: Updated it!

File: src/locales/en/common.json [hunk 3]
```diff
@@ -354,6 +354,7 @@
     "your_balance": "Your Balance",
     "volume_24": "Volume 24h",
     "or": "or"
+
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/locales/en/common.json]: Removed extra space.

File: src/selectors/appSettings.js [hunk 4]
```diff
@@ -48,7 +47,7 @@ export function useBiometricsSelector() {
 export const maxPinCodeLengthSelector: Selector<number> = createSelector(
   ({ appSettings }) => appSettings,
   ({ data: appSettings }: AppSettingsReducerState): number => {
-    if (appSettings?.hasSixDigitsPin) return 6;
+    if (appSettings?.hasSixDigitsPin) return 4;
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/selectors/appSettings.js]: Commented this line.

**Pull request merged**